crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -51,6 +51,16 @@ def create_proposal
   end
 
   def validate_proposal_after_save proposal
+    proposal["attributes"]["provisioner"]["packages"].each do |packages_for_platform|
+      platform_version = packages_for_platform[0]
+      platform_packages = packages_for_platform[1]
+      platform_packages.each do |package|
+        unless package =~ /(^\w[\w\+\.-<=>]+$)/
```

Comment:
> Underscore is a \w word character.

Yes, you are right, sorry.

> The position of the dot doesn't matter. It's meant as the character \. and not as /./ ... or do you ask for this just for cosmetic reasons?

I'm not talking about the dot, just about the dash. It is mostly safer to put it at the end to avoid some bad behavior like taking the dash for a rang instead of matching for a dash :)